From f9604f5c46ad262cc2ef2426e8695bb2c156e5fa Mon Sep 17 00:00:00 2001
From: David Heidelberg <david@ixit.cz>
Date: Wed, 17 Apr 2024 12:27:51 +0100
Subject: [PATCH] hack! arm64: dts: qcom: sdm845: disable qcrypto

The qcrypto driver has regressed on SDM845, attempts to use it
may result in an instant hard-crash and the device either resets or
goes to crashdump mode (depending on vendor configuration).

Disable it until this is resolved.

[David] I haven't seen related crash when testing, but some may still
occur somewhere.

Original author: Casey Connolly <casey.connolly@linaro.org>

Signed-off-by: David Heidelberg <david@ixit.cz>
---
 arch/arm64/boot/dts/qcom/sdm845.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sdm845.dtsi b/arch/arm64/boot/dts/qcom/sdm845.dtsi
index daf63392a078..4ccd95038d41 100644
--- a/arch/arm64/boot/dts/qcom/sdm845.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845.dtsi
@@ -2776,6 +2776,12 @@ crypto: crypto@1dfa000 {
 				 <&apps_smmu 0x706 0x1>,
 				 <&apps_smmu 0x714 0x1>,
 				 <&apps_smmu 0x716 0x1>;
+
+			/*
+			 * FIXME: disabled due to a regression causing a trap to TZ
+			 * and a hard reset if this device is used.
+			 */
+			status = "disabled";
 		};
 
 		ipa: ipa@1e40000 {
-- 
GitLab

